src/hipscat/catalog/pixel_node.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
```

Comment:
Does this import get used?

Reply:
Yes. In python < 3.10 it's necessary to allow a type hint for the class currently being defined (e.g. in line 24 where the init has a type referencing its own class)